staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -323,6 +327,10 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 	}
 
 	if err := cmd.Run(); err != nil {
+		if a.installHint != "" {
+			return fmt.Errorf("exec: %v\n%s", err, cleanANSIEscapeCodes(a.installHint))
```

Comment:
We need to make the overall error message more useful too (i.e. tell the user what an exec credential plugin is / point them to the docs, etc).

Reply:
I need to spend some more time looking at other error messages printed by `kubectl` for status quo. Until I do that, I propose this error message format as a first draft. WDYT?

One thing that worries me is that if this error message were to show up in JSON logs for a non-`kubectl` client, they would look pretty ugly :(

With `InstallHint`:
```
Unable to connect to the server: getting credentials: exec: executable "does-not-exist" not found

Looks like you are trying to use a client-go credential plugin that is not installed.

To learn more about this feature, check out the following link.
      https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins

example-client-go-exec-plugin is required to authenticate
to the current cluster.  It can be installed:

On macOS: brew install example-client-go-exec-plugin

On Ubuntu: apt-get install example-client-go-exec-plugin

On Fedora: dnf install example-client-go-exec-plugin
```

Without `InstallHint` (i.e., `InstallHint == ""`).
```
Unable to connect to the server: getting credentials: exec: executable "does-not-exist" not found

Looks like you are trying to use a client-go credential plugin that is not installed.

To learn more about this feature, check out the following link.
      https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins
```

Reply:
I implemented this in https://github.com/kubernetes/kubernetes/pull/91305/commits/e53aca0c6f8c9a2c82abafd2e7baf6bd0b1c894e. Not wed to it.